### PR TITLE
Allow plans to have null concentration

### PR DESCRIPTION
**Pull request by Nathan-Lackie**

### CODE REVIEW

**Comment by Nathan-Lackie** [comment]
Resolves #77

File: src/api/plans.ts [hunk 1]
```diff
@@ -36,7 +36,7 @@ router.get("/plans", async (req, res) => {
   >`SELECT user_plans.id, graduation_date, programs.name AS program_name, concentrations.name AS concentration_name
     FROM user_plans
     JOIN programs ON programs.id = program_id
-    JOIN concentrations ON concentrations.id = concentration_id
+    LEFT JOIN concentrations ON concentrations.id = concentration_id
```
Comment (KaiSnorlax) [inline src/api/plans.ts:38]: erm please fix this formatting.

File: src/api/plans.ts [hunk 2]
```diff
@@ -52,14 +52,14 @@ router.get("/plans", async (req, res) => {
...
-  concentrationId: z.number(),
+  concentrationId: z.number().nullable().optional(),
 });
 router.post("/plans", async (req, res) => {
   const user = await authUser(req);
   const planBody = planReqBody.parse(req.body);
 
   const planId = await one<{ id: number }>(
-    sql`INSERT INTO user_plans (user_id, graduation_date, concentration_id, program_id) VALUES (${user.id}, ${planBody.graduationDate}, ${planBody.concentrationId}, ${planBody.programId}) RETURNING id`,
+    sql`INSERT INTO user_plans (user_id, graduation_date, concentration_id, program_id) VALUES (${user.id}, ${planBody.graduationDate}, ${planBody.concentrationId ?? null}, ${planBody.programId}) RETURNING id`,
```
Comment (KaiSnorlax) [inline src/api/plans.ts]: this is awkward... please fix this formatting.

File: src/api/plans.ts (hunk 1 above)
Comment (Nathan-Lackie) [reply src/api/plans.ts:38]: Out of scope for this PR, existing issue

File: src/api/plans.ts (hunk 2 above)
Comment (Nathan-Lackie) [reply src/api/plans.ts]: Out of scope for this PR, existing issue

File: src/api/plans.ts (hunk 1 above)
Comment (KaiSnorlax) [reply src/api/plans.ts:38]: me when i need to clean my room

**Review by KaiSnorlax — APPROVED** [review]
lgtm

**Pull request merged**